libs/agno/agno/agent/agent.py

```diff
@@ -668,6 +668,12 @@ def _run(
                         # Format tool calls whenever new ones are added during streaming
                         self.run_response.formatted_tool_calls = format_tool_calls(self.run_response.tools)
 
+                        for tool_call in tool_calls_list:
+                            tool_name = tool_call.get("tool_name", "")
+                            if tool_name.lower() in ["think", "analyze"]:
+                                tool_args = tool_call.get("tool_args", {})
+                                self.update_reasoning_content_from_tool_call(tool_name, tool_args)
+
```

Comment:
I think we need to do this on the tool_call_completed stage? Or just after both stream and non-stream?

Reply:
done